### DISCUSSION TITLE
Github Actions Workflow metadata

### DISCUSSION THREAD

**Discussion by msuozzo** [community/community#4188] [Actions]
## Current State

The current Actions [context metadata](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context) contains the following field:

`github.workflow` | `string` | The name of the workflow. If the workflow file doesn't specify a name, the value of this property is the full path of the workflow file in the repository.
-- | -- | --

This is the only data provided on the source workflow and seems mainly oriented at descriptive, human-readable contexts. The workflow path is useful on its own but is only conditionally available on `workflow_dispatch` events under the `github.event.workflow` key.

## Desired State

If possible, it would be useful to break this out into two separate fields:

* [existing] `workflow` or `workflow_name`: same as `github.workflow` and would still default to the path value if no name were provided.
* [new] `workflow_path`: unconditionally the full path of the executed workflow.

This new path field would allow the action to provide a helpful self-link either to the console output or to a self-describing artifact e.g. an audit log.

**Comment by rajbos** [comment]
Addtional fields are welcomed as well: I would like to add or change the text show in the workflow run, for example [here](https://github.com/rajbos/dependency-updates/actions/workflows/pipeline.yml) or add additional info on top of the workflow name. 

For example: we have a different workflow to push to production, and would like to indicate the name of the environment being used for the execution.

**Comment by joao-paulo-parity** [comment]
Chiming in with one more use-case: @paritytech needs the `workflow_path` for https://github.com/paritytech/pr-custom-review in order to enforce that the action is created from a workflow in a specific path. This enforcement helps in securing the workflow's code against deletions or other changes which could modify it without approval from its owners. Further explanation is given in https://github.com/paritytech/pr-custom-review/issues/69.

**Reply by jsoref** [reply]
I'd really love easy access to the `workflow_path`, but it's possibly to get it:
https://github.com/check-spelling/check-spelling/blob/610473ebb9d729a89c00db1b0d5eec3bfd5afc68/unknown-words.sh#L279-L292
